### PR TITLE
Add subtitle persistence on iOS and Android

### PR DESCRIPTION
**Pull request by elv-peter**
iOS now works as expected, except: If there is a saved subtitle id when a video loads, the correct captions are enabled, but the Settings UI will still display "off".

On Android, videos will initially load with subtitles off, but the Settings UI will display "eng" or whatever language Bitmovin automatically tried to turn on.

I considered removing a lot of the fullscreen code, since it seemed to work without it, but refrained in case it was to address some corner cases.

### CODE REVIEW

File: android/src/main/java/com/xxsnakerxx/RNBitmovinPlayer/RNBitmovinPlayerManager.java [hunk 1]
```diff
@@ -217,136 +236,111 @@ public void setConfiguration(BitmovinPlayerView view, ReadableMap config) throws
...
-    {
-        this._decorView.post(new Runnable()
-        {
-            @Override
-            public void run()
-            {
-                int uiParams = FullscreenUtil.getSystemUiVisibilityFlags(fullScreen, true);
-                _decorView.setSystemUiVisibility(uiParams);
-            }
+    //private void doRotation(boolean fullScreen) {
```
Comment (elv-serban) [inline android/src/main/java/com/xxsnakerxx/RNBitmovinPlayer/RNBitmovinPlayerManager.java:319]: Was doRotation() not used?

Comment (elv-peter) [reply android/src/main/java/com/xxsnakerxx/RNBitmovinPlayer/RNBitmovinPlayerManager.java:319]: Correct

File: android/src/main/java/com/xxsnakerxx/RNBitmovinPlayer/RNBitmovinPlayerManager.java [hunk 2]
```diff
@@ -170,41 +190,40 @@ public void onDropViewInstance(BitmovinPlayerView view) {
...
+        String sourceUrl;
+        if (sourceMap.hasKey("dash")) {
+            sourceUrl = sourceMap.getString("dash");
+        } else if (sourceMap.hasKey("hls")) {
+            sourceUrl = sourceMap.getString("hls");
+        } else {
+            System.out.println("RNBitmovinPlayerManager.setConfiguration - Could not find dash or hls in \"Source\"");
+            return;
         }
+        System.out.println("RNBitmovinPlayerManager.setConfiguration - sourceUrl=" + sourceUrl);
```
Comment (elv-wayne) [inline android/src/main/java/com/xxsnakerxx/RNBitmovinPlayer/RNBitmovinPlayerManager.java:212]: We probably don't want to print this for production

File: android/src/main/java/com/xxsnakerxx/RNBitmovinPlayer/RNBitmovinPlayerManager.java (hunk 1 above)
Comment (elv-wayne) [reply android/src/main/java/com/xxsnakerxx/RNBitmovinPlayer/RNBitmovinPlayerManager.java:319]: I started doing native full screen based on the examples but found a way to rotate using react-native side.
System Visibility affects different android versions and tablets so I left it in there (takes away system nav bars etc).

File: android/src/main/java/com/xxsnakerxx/RNBitmovinPlayer/RNBitmovinPlayerManager.java [hunk 3]
```diff
@@ -217,136 +236,111 @@ public void setConfiguration(BitmovinPlayerView view, ReadableMap config) throws
...
-        {
-            updateLayoutRunnable.run();
-        }
-        else
-        {
-            Handler handler = new Handler(mainLooper);
-            handler.post(updateLayoutRunnable);
-        }
+    private void doLayoutChanges(final boolean fullscreen) {
+        _decorView.post(new UpdateLayoutRunnable(fullscreen));
```
Comment (elv-wayne) [inline android/src/main/java/com/xxsnakerxx/RNBitmovinPlayer/RNBitmovinPlayerManager.java:343]: Hopefully this code works on all android devices.

**Review by elv-wayne — APPROVED** [review]
Should retest all functions including full screen with as many simulators and devices as possible with these changes.

File: android/src/main/java/com/xxsnakerxx/RNBitmovinPlayer/RNBitmovinPlayerManager.java (hunk 3 above)
Comment (elv-peter) [reply android/src/main/java/com/xxsnakerxx/RNBitmovinPlayer/RNBitmovinPlayerManager.java:343]: `doSystemUiVisibility` used the same method to run on UI thread and it's called at the same time so I think it should be OK.

Actually I tried not calling either function and fullscreen still worked for me

**Pull request merged**